src/openepd/api/dto/common.py

```diff
@@ -0,0 +1,114 @@
+#
+#  Copyright 2023 by C Change Labs Inc. www.c-change-labs.com
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  This software was developed with support from the Skanska USA,
+#  Charles Pankow Foundation, Microsoft Sustainability Fund, Interface, MKA Foundation, and others.
+#  Find out more at www.BuildingTransparency.org
+#
+import abc
+import datetime
+from typing import Final, Generic, TypeAlias, TypeVar
+
+import pydantic as pyd
+from pydantic.generics import GenericModel
+
+from openepd.api.dto.base import BaseMetaDto, BaseOpenEpdApiModel, MetaExtensionBase
+from openepd.api.dto.meta import PerformanceMetaMixin
+from openepd.model.base import AnySerializable, BaseOpenEpdSchema
+
+DEFAULT_PAGE_SIZE: Final[int] = 100
+MAX_PAGE_SIZE: Final[int] = 250
+
+
+class AuditableDto(BaseOpenEpdApiModel, metaclass=abc.ABCMeta):
```

Comment:
Do you think we'll be able to replace the local version of same with this? E.g. the serverside.